torch/onnx/_internal/fx/serialization.py

```diff
@@ -126,30 +128,27 @@ def save_model_with_external_data(
             # to "transformer_attention_self_query_weight" for mimicking the
             # name-modifying code in FX-to-ONNX exporter.
             # See function _replace_get_attr_with_placeholder for details.
-            refined_name = name.replace(".", "_")
-
-            # For each refined PyTorch tensor name loaded by torch.load,
+            name = name.replace(".", "_") if rename_initializer else name
+            # For each PyTorch tensor name loaded by torch.load,
             #  1.  Search its best match in ONNX model. E.g., the match of
             #       "transformer_attention_weight" could be "attention_weight".
             #  2.  Set "tensor" as the initializer of the matched ONNX input.
             #      E.g., "tensor" is stored as the initializer of "attention_weight".
             # Step 1 is required because sometimes, tensor names are stored with prefix the dictionary
```

Comment:
omg another hack

I believe this is not needed for dynamo path fake export.

Reply:
In fact I think this logic can be put under `rename_initializer` flag too.

Reply:
good catch